### PR TITLE
[v6.2] docs: improve go API example

### PR DESCRIPTION
**Pull request by inertial-frame**
This PR addresses part of the new refactoring and layout improvements for: https://github.com/gravitational/teleport/pull/6694:

1. When I walk through the v6.2 API example as is, I encounter:

Screen Shot 2021-05-14 at 12 53 20 PM

2. I also encounter:

Screen Shot 2021-05-14 at 12 53 47 PM

As is the API client will hang indefinitely without displaying any message breaking the flow of the article.

3. The addition of the `--proxy` flag and `Addrs: []string` entries appropriate for one's configuration now correctly displays the following:

Screen Shot 2021-05-14 at 12 54 23 PM

4. I also improved the log outputs to display the `response` object which I think is useful.

### CODE REVIEW

**Comment by inertial-frame** [comment]
Note for reviewers: I've also combined several changes that were originally made for the v6.2 Server Access documentation. We will likely briefly describe some of the API-related topics within the Architecture overview within that section and direct readers to this article for an actual example. Thanks!

File: docs/pages/reference/api/getting-started.mdx [hunk 1]
```diff
@@ -1,49 +1,54 @@
...
-## 2/3 Generate Client Credentials
+## Step 2/3. Generate client credentials
 
-Login as the newly created user with `tsh`.
+Log in as the newly created user with `tsh`.
 
 ```bash
 # generate tsh profile
-tsh login --user=api-user
+sudo tsh login --user=api-admin --proxy=tele.example.com:443
```
Comment (klizhentas) [inline docs/pages/reference/api/getting-started.mdx:45]: let's remove sudo from `tsh`

File: docs/pages/reference/api/getting-started.mdx [hunk 2]
```diff
@@ -1,49 +1,54 @@
...
+  Read [API authorization](./architecture.mdx#authorization) to learn more about defining custom roles for your API client.
+</Admonition>
+
+Create a user `api-admin` with the built-in role `admin`:
 
 ```bash
 # Run this directly on your auth server
 # Add user and login via web proxy
-tctl users add api-user --roles=admin
+sudo tctl users add api-admin --roles=admin
```
Comment (klizhentas) [inline docs/pages/reference/api/getting-started.mdx:36]: sudo is not always necessary for `tctl`, for example it's not needed for cloud

**Review by klizhentas — COMMENTED** [review]
some tweaks, otherwise LGTM @Joerger can you also take a look

File: docs/pages/reference/api/getting-started.mdx [hunk 3]
```diff
@@ -69,31 +75,43 @@ func main() {
...
+	
+	log.Printf("Example success!")
+	log.Printf("Example server response: %s", resp)
+	log.Printf("Server version: %s", resp.ServerVersion)
 }
+EOF
 ```
 
-Now you can run the program which will connect the client to the Teleport Auth server and fetch the server version.
+Now you can run the program and connect the client to the Teleport Auth server and fetch the server version.
```
Comment (inertial-frame) [inline docs/pages/reference/api/getting-started.mdx:106]: Note to self:  `and fetch the server version.` to ` to fetch the server version.

File: docs/pages/reference/api/getting-started.mdx (hunk 1 above)
Comment (inertial-frame) [reply docs/pages/reference/api/getting-started.mdx:45]: Done.

File: docs/pages/reference/api/getting-started.mdx [hunk 4]
```diff
@@ -69,31 +75,43 @@ func main() {
 	ctx := context.Background()
 
 	clt, err := client.New(ctx, client.Config{
+		Addrs: []string{
```
Comment (Joerger) [inline docs/pages/reference/api/getting-started.mdx:78]: `Addrs` aren't necessary when you use `client.LoadProfile`. The proxy address used to login via `tsh`, in this case `tele.example.com:443`, will be used to retrieve the auth address. No harm in listing them if you think it serves a purpose, just wanted to point it out.

**Review by Joerger — APPROVED** [review]

File: docs/pages/reference/api/getting-started.mdx (hunk 4 above)
Comment (inertial-frame) [reply docs/pages/reference/api/getting-started.mdx:78]: I just tested with and without the `Addrs` array. All other things, including `-proxy` flag being left the same:

With it:
```
go run main.go

2021/05/24 20:08:41 Example success!
2021/05/24 20:08:41 Example server response: {publictest.go-teleport.com 6.1.5 Kubernetes:true App:true DB:true  publictest.go-teleport.com:443 {}  %!s(int32=0)}
2021/05/24 20:08:41 Server version: 6.1.5
```

Without it:
```
go run main.go
2021/05/24 20:10:41 failed to create client: all auth methods failed
	context deadline exceeded
exit status 1
```

Comment (inertial-frame) [reply docs/pages/reference/api/getting-started.mdx:78]: The Teleport server itself generates the following message:

```
ERRO [PROXY:SER] "proxy2021/05/24 20:08:41 http: TLS handshake error from 3.140.251.104:43494: acme can't get a cert for domain teleport.cluster.local, add it to the proxy_service.public_addr, or use one of the domains: publictest.go-teleport.com\n" utils/cli.go:304
```

Comment (Joerger) [reply docs/pages/reference/api/getting-started.mdx:78]: Hmm that should work, maybe the proxy address you used to log in is different from the ones you listed in `Addrs`.

If you have time, could you make a bug ticket? Include your steps for logging in and the contents of `~/.tsh/current-profile` and `~/.tsh/<current-profile>.yaml`.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**